src/lib/operators/join_hash/hash_functions.hpp

```diff
@@ -0,0 +1,556 @@
+#pragma once
```

Comment:
Haven't thought about it that way. You're right.

I think `hash_helpers` doesn't help as this file contains both `build` and `probe` and thus pretty much the whole hash join.
Any ideas?

Reply:
Please remind me why this was moved into its own file. To enable testing on those functions?

Reply:
Yes, we don't want them in the main hpp file due to compile time reasons and testing them from the CPP file would be possible but only via including the file in the tests.
Besides, I even think it's better know because we separate process flow from the actual implementation (IMO).

Reply:
Got it. A comment at the top of the file would be great.

> Besides, I even think it's better know because we separate process flow from the actual implementation (IMO).

I get that point, but then it also forces people to jump between two files to follow the code.

Anyway, how about `join_hash_steps.hpp`?